src/main/java/com/laytonsmith/core/MethodScriptCompiler.java

```diff
@@ -1575,14 +1575,50 @@ public static ParseTree compile(TokenStream stream) throws ConfigCompileExceptio
 
 		assert t != null;
 
+		// Handle mismatching square brackets "[]".
+		assert arrayStack.size() != 0 : "The last element of arrayStack should be present, but it was popped.";
 		if(arrayStack.size() != 1) {
-			throw new ConfigCompileException("Mismatched square brackets", t.target);
+
+			// Some starting square bracket '[' was not closed at the end of the script.
+			// Find the last '[' that was not closed and use that as target instead of the last line of the script.
+			Target target = traceMismatchedOpenToken(stream, TType.LSQUARE_BRACKET, TType.RSQUARE_BRACKET);
+			assert target != null : "Mismatched bracket was detected, but target-finding code could not find it.";
+			if(target == null) {
```

Comment:
This isn't possible? You just asserted that target is not null, so target cannot be null at this point.